src/main/java/core/basesyntax/impl/PurchaseHandlerImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.impl;
+
+import dao.FruitDao;
+import dao.FruitDaoImpl;
+import model.FruitTransaction;
+import strategy.OperationHandler;
+
+public class PurchaseHandlerImpl implements OperationHandler {
+    private FruitDao fruitDao = new FruitDaoImpl();
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        int quantityBefore = fruitDao.get(transaction.getFruit());
+        if (quantityBefore < transaction.getQuantity()) {
+            throw new RuntimeException("There is not enough fruit in the storage");
+        }
+        fruitDao.add(transaction.getFruit(), quantityBefore - transaction.getQuantity());
+    }
+}
```

Comment:
It's better to use dependency injection for the `FruitDao` instance instead of creating a new object inside the `PurchaseHandlerImpl` class. This will make your code more testable and adherent to the SOLID principles, particularly the Dependency Inversion Principle.